osc-server/src/test/java/org/osc/core/broker/service/servicefunctionchain/UpdateServiceFunctionChainServiceTest.java

```diff
@@ -60,91 +60,77 @@ public void testDispatch_WithNullRequest_ThrowsNullPointerException() throws Exc
 		
+		//Arrange
 		BaseDto dto = new BaseDto();
 		dto.setId(this.sfc.getId());
 		dto.setParentId(this.vc.getId());
 		request.setDto(dto);
 		request.setName(this.sfc.getName());
 		List<Long> vsIds = new ArrayList<Long>();
+		
 		vsIds.add(this.vs.getId());
 		vsIds.add(this.vs1.getId());
 		request.setVirtualSystemIds(vsIds);
 		Mockito.when(this.service.validator.validateAndLoad(request)).thenReturn(this.sfc);
-		// Act.
-		BaseJobResponse response = this.service.dispatch(request);
-		Assert.assertNotNull("The returned response should not be null.", response);
-	}
-	
-	@Test
-	public void testDispatch_sfcVirutalSystemIdsInOrder_ValidationSucceeds() throws Exception {
 		
-		BaseDto dto = new BaseDto();
-		dto.setId(this.sfc.getId());
-		dto.setParentId(this.vc.getId());
-		request.setDto(dto);
-		request.setName(this.sfc.getName());
-		List<Long> vsIds = new ArrayList<Long>();
-		vsIds.add(this.vs1.getId());
-		vsIds.add(this.vs.getId());
-		request.setVirtualSystemIds(vsIds);
-		Mockito.when(this.service.validator.validateAndLoad(request)).thenReturn(this.sfc);
 		// Act.
-		//update sfc with virtual system ids list {2,1}
 		BaseJobResponse response = this.service.dispatch(request);
 		
-		//check Virtual system is as updated
-		List<Long> sfcVsIdList = this.em.find(ServiceFunctionChain.class, this.sfc.getId()).
-				getVirtualSystems().stream().map(vss -> vss.getId()).collect(Collectors.toList());
-		//read back sfc and cross check with local Virtual system list for order
-		Assert.assertEquals(vsIds, sfcVsIdList);
+		//Assert
 		Assert.assertNotNull("The returned response should not be null.", response);
 	}
 	
 	@Test
-	public void testDispatch_sfcVirutalSystemIdsNotInOrder_ValidationFails() throws Exception {
+	public void testDispatch_validSFCVirtualSystemIds_SFCIsUpdated() throws Exception {
 		
+		//Arrange
 		BaseDto dto = new BaseDto();
 		dto.setId(this.sfc.getId());
 		dto.setParentId(this.vc.getId());
 		request.setDto(dto);
 		request.setName(this.sfc.getName());
 		List<Long> vsIds = new ArrayList<Long>();
+		
 		vsIds.add(this.vs1.getId());
 		vsIds.add(this.vs.getId());
 		request.setVirtualSystemIds(vsIds);
 		Mockito.when(this.service.validator.validateAndLoad(request)).thenReturn(this.sfc);
+		
 		// Act.
 		//update sfc with virtual system ids list {2,1}
 		BaseJobResponse response = this.service.dispatch(request);
 		
 		//check Virtual system is as updated
 		List<Long> sfcVsIdList = this.em.find(ServiceFunctionChain.class, this.sfc.getId()).
 				getVirtualSystems().stream().map(vss -> vss.getId()).collect(Collectors.toList());
-		
-		//change the order to compare
-		vsIds.clear();
-		vsIds.add(this.vs.getId());
-		vsIds.add(this.vs1.getId());
 		//read back sfc and cross check with local Virtual system list for order
-		Assert.assertNotEquals(vsIds, sfcVsIdList);
+		
+		//Assert
+		Assert.assertEquals("The list of virtual system ids is different than expected", vsIds, sfcVsIdList);
```

Comment:
Is this test the same as the previous one but with additional validation? If so you should have just a single test case containing all the validation/asserts that are applicable. You would have a new test case only if the inputs, action or mocked state changes but you can pack as much validation as possible on the same test method/case

Reply:
Added three test cases with Empty list of Virtual systems, a updated virtual systems and last one is Validation of the order. Yes the last two are similar, but do not want to stub into everything as I change the input list to check back the action.